examples/custom-tpu-send-transactions/src/main.rs

```diff
@@ -1,3 +1,519 @@
+        finalized_block_information.block_height,
+        finalized_block_information.slot,
+        finalized_block_information.blockhash
+    );
+    println!(
+        "From RPC blockheight : {block_height:?}, hash: {}",
+        blockhash
+    );
+
+    let finalized_block_information = BlockInformation {
+        slot: finalized_block_information.slot,
+        block_height,
+        last_valid_blockheight: finalized_block_information.block_height + 300,
+        cleanup_slot: finalized_block_information.slot + 1000000,
+        blockhash: finalized_block_information.blockhash,
+        commitment_config: CommitmentConfig::finalized(),
+        block_time: 0,
+    };
+
+    let block_information_store = BlockInformationStore::new(finalized_block_information);
+
+    let data_cache = DataCache {
+        block_information_store,
+        cluster_info: ClusterInfo::default(),
+        identity_stakes: IdentityStakes::new(validator_identity.pubkey()),
+        slot_cache: SlotCache::new(finalize_slot),
+        tx_subs: SubscriptionStore::default(),
+        txs: TxStore {
+            store: Arc::new(DashMap::new()),
+        },
+        epoch_data: EpochCache::new_for_tests(),
+        leader_schedule: Arc::new(RwLock::new(CalculatedSchedule::default())),
+    };
+
+    let data_cache_service = DataCachingService {
+        data_cache: data_cache.clone(),
+        clean_duration: Duration::from_secs(120),
+    };
+
+    // start listning the cluster data and filling the cache
+    data_cache_service.listen(
+        endpoints.blocks_notifier.resubscribe(),
+        endpoints.blockinfo_notifier,
+        endpoints.slot_notifier.resubscribe(),
+        endpoints.cluster_info_notifier,
+        endpoints.vote_account_notifier,
+    );
+
+    let count = args.transaction_count.unwrap_or(10);
+    let prioritization_heap_size = Some(count * args.number_of_seconds);
+
+    let tpu_config = TpuServiceConfig {
+        fanout_slots: args.fanout_slots.unwrap_or(16),
+        maximum_transaction_in_queue: 2000000,
+        quic_connection_params: QuicConnectionParameters {
+            connection_timeout: Duration::from_secs(60),
+            connection_retry_count: 10,
+            finalize_timeout: Duration::from_millis(10000),
+            max_number_of_connections: 4,
+            unistream_timeout: Duration::from_millis(1000),
+            write_timeout: Duration::from_secs(10),
+            number_of_transactions_per_unistream: 1,
+            unistreams_to_create_new_connection_in_percentage: 5,
+            prioritization_heap_size,
+        },
+        tpu_connection_path: TpuConnectionPath::QuicDirectPath,
+    };
+
+    let tpu_service: TpuService = TpuService::new(
+        tpu_config,
+        validator_identity,
+        leader_schedule,
+        data_cache.clone(),
+    )
+    .await?;
+    let transaction_service_builder = TransactionServiceBuilder::new(
+        TxSender::new(data_cache.clone(), tpu_service.clone()),
```

Comment:
the services are quite tangled. shouldn't it be easy to just wire up the TxSender? it requires recent blockhash and leader schedule

leader schedule could be:
1. full schedule from RPC
2. schedule for current epoch as text-file with each lead qualified by slots
3. hardcoded leaders (if the setup is just 1 test-validator or a small number of test nodes)